### PR TITLE
Ansibilize Horton Bot

### PR DESCRIPTION
**Pull request by mcritchlow**
HI JOHN

Ok here's an initial working set of playbooks for deploying the bot. Some notes below:

1. I'm targeting `master` in `bot.yml`, so to test you'll need to pass in this branch as follows:

`$ ansible-playbook setup.yml --ask-vault-pass --extra-vars "ansistrano_git_branch=ansibilize"`

2. In `before_symlink` you'll see i have `become: no` which I know is normally redundant. But in this case the hook is inheriting `yes` upstream and so I had to explicitly set it to `no` here so the vagrant user could access it's installed `bundler` binary

3. In `puma.service` I'm passing in the slack token via `ENVIRONMENT`. I'm especially curious what you think about this. It works, but is obviously writing the token into the service file. If there's a better place/way to set the variable let me know.

4. The vault password is in LastPass in `Shared-DevOps/Library Slack Bot - Vault Password`

Fixes: #24 #5

### CODE REVIEW

File: ops/deploy.yml [hunk 1]
```diff
@@ -3,14 +3,13 @@
...
     - vars/slack_token.yml
 
   roles:
     - { role: carlosbuenosvinos.ansistrano-deploy, become: yes }
 
-  tasks:
+  pre_tasks:
     - name: Get timestamp for repo
       hosts: localhost
       shell: 'date +%Y%m%d%H%M%S.%5N'
```
Comment (jhriv) [inline ops/deploy.yml]: add a "changed_when: false" here

File: ops/vars/ansistrano.yml [hunk 2]
```diff
@@ -13,7 +13,8 @@ ansistrano_shared_paths:
 
 ansistrano_deploy_via: "git"
 ansistrano_git_repo: https://github.com/ucsdlib/horton-bot.git
-ansistrano_git_branch: master
+# TODO: change to master before deployment.
+ansistrano_git_branch: ansibilize
```
Comment (jhriv) [inline ops/vars/ansistrano.yml]: Pretty sure `ansistrano_git_branch` will need to change _before_  the PR gets merged

File: .ruby-version [hunk 3]
```diff
@@ -1 +1 @@
-2.4.1
+2.4.2
```
Comment (jhriv) [inline .ruby-version]: Has this hot fix been applied to the master branch yet?

**Comment by jhriv** [comment]
The only show stoppers I have seen:

1. the `ansistrano_git_branch` pointing to not-master.
2. Using `hosts:` in a play is ignored. You need `connection: local` to prevent running on the remote host, and the `run_once: true` to ensure the timestamp is generated _only once_ even when pushing to multiple hosts.

Placing `become:` between  ` name:` and the module itself is personally distasteful to me, but not enough of a show stopper to demand fixing.

This PR also bumps ruby to 2.4.2, which I don't think is in the master branch yet.

File: .ruby-version (hunk 3 above)
Comment (mcritchlow) [reply .ruby-version]: Nope, it would come in as part of this PR. Why do you ask?

**Comment by mcritchlow** [comment]
1. I think it is pointing to `master` already? https://github.com/ucsdlib/horton-bot/pull/51/files#diff-e32d9925e6709cfcec416425566787f6R28

2. Will to. Tbh, I pulled the initial version of this from the horton/notch8 playbook.. so eek. But I'll update it!

**Comment by mcritchlow** [comment]
@jhriv - let me know if https://github.com/ucsdlib/horton-bot/pull/51/commits/a6c5a1fe3ac6ed1b7b6f38e67a6dcfedbcaf18d5 addresses the issues with that task. Thanks!

**Comment by mcritchlow** [comment]
@jhriv - any other changes you want to see here? If not, merge please? :)

**Pull request merged**